File: model/src/main/java/org/deep_thinker/model/DQNConfig.java
package org.deep_thinker.model;

public class DQNConfig {
    public float startEpsilon;
    public float endEpsilon;
    public float explorationFraction;
    public int learningStarts;
    public int trainFrequency;
    public int batchSize;
    public float learningRate;
    public int targetNetworkFrequency;
    public float gamma;
    public int numInputs;
    public int numActions;
    public int hidden1Size;
    public int hidden2Size;
    public int totalTimesteps;
    public int replayBufferSize;

    public DQNConfig(float startEpsilon,
                     float endEpsilon,
                     float explorationFraction,
                     int learningStarts,
                     int trainFrequency,
                     int batchSize,
                     float learningRate,
                     int targetNetworkFrequency,
                     float gamma,
                     int numInputs,
                     int numActions,
                     int hidden1Size,
                     int hidden2Size,
                     int totalTimesteps,
                     int replayBufferSize) {
        this.startEpsilon = startEpsilon;
        this.endEpsilon = endEpsilon;
        this.explorationFraction = explorationFraction;
        this.learningStarts = learningStarts;
        this.trainFrequency = trainFrequency;
        this.batchSize = batchSize;
        this.learningRate = learningRate;
        this.targetNetworkFrequency = targetNetworkFrequency;
        this.gamma = gamma;
        this.numInputs = numInputs;
        this.numActions = numActions;
        this.hidden1Size = hidden1Size;
        this.hidden2Size = hidden2Size;
        this.totalTimesteps = totalTimesteps;
        this.replayBufferSize = replayBufferSize;
    }
}

File: model/src/main/kotlin/org/deep_thinker/model/DQNConfig.kt
package org.deep_thinker.model

data class DQNConfig(
    val startEpsilon: Float,
    val endEpsilon: Float,
    val explorationFraction: Float,
    val learningStarts: Int,
    val trainFrequency: Int,
    val batchSize: Int,
    val learningRate: Float,
    val targetNetworkFrequency: Int,
    val gamma: Float,
    val numInputs: Int,
    val numActions: Int,
    val hidden1Size: Int,
    val hidden2Size: Int,
    val totalTimesteps: Int,
    val replayBufferSize: Int
)

